datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -489,4 +495,20 @@ private String getAccountIdFromCrn(String userCrn) {
     private DetailedEnvironmentResponse getEnvironment(SdxClusterRequest sdxClusterRequest) {
         return environmentEndpoint.getByName(sdxClusterRequest.getEnvironment());
     }
+
+    public void save(SdxCluster sdxCluster) {
+        sdxClusterRepository.save(sdxCluster);
+    }
+
+    public boolean hasActiveFlow(SdxCluster sdxCluster) {
+        try {
+            List<FlowLogResponse> flowLogsByResourceNameAndChainId =
+                    flowEndpoint.getFlowLogsByResourceNameAndChainId(sdxCluster.getClusterName(), sdxCluster.getRepairFlowChainId());
+            return flowLogsByResourceNameAndChainId.stream()
+                    .anyMatch(flowLog -> flowLog.getStateStatus().equals(StateStatus.PENDING) || !flowLog.getFinalized());
+        } catch (Exception e) {
+            LOGGER.error("Exception occured during getting flow logs from CB: {}", e.getMessage());
+            return false;
+        }
+    }
```

Comment:
This is working only for repair, you cannot use it to check start or stop flow, because it is checking active flows based on `sdxCluster.getRepairFlowChainId()`

Reply:
I reverted to the previous version